### PR TITLE
Fargate cluster type

### PR DESCRIPTION
**Pull request by faabiosr**
Was defined a cluster_type property for creating an specific ecs cluster and added variable for creating task definitions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +79,7 @@ Including an example of how to use your role (for instance, with variables passe
               - us-east-1a
             subnets:
               - subnet-abcd1234
+            custer_type: ec2
```
Comment (mtulio) [inline README.md]: `cluster_type`

File: tasks/cluster.yml [hunk 2]
```diff
@@ -5,13 +5,16 @@
 
 - set_fact:
     cluster_userdata: "{{ lookup('template', 'lcg_userdata.j2') }}"
+  when: item_c.cluster_type == "ec2"
```
Comment (mtulio) [inline tasks/cluster.yml]: Too many conditional (`when`) here, IMHO you could use one `block` here. What do you think?

**Review by mtulio — CHANGES_REQUESTED** [review]
Great improvement, a few changes.

File: README.md (hunk 1 above)
Comment (fabiorphp) [reply README.md]: done!

File: tasks/cluster.yml (hunk 2 above)
Comment (fabiorphp) [reply tasks/cluster.yml]: done!

File: tasks/cluster.yml [hunk 3]
```diff
@@ -3,40 +3,43 @@
...
+      launch_config_name: "{{ item_c.name }}-{{ ansible_date_time.epoch }}"
+      min_size: "{{ item_c.asg_min_size }}"
+      max_size: "{{ item_c.asg_max_size }}"
+      desired_capacity: "{{ item_c.asg_desired_size }}"
+      region: "{{ item_c.region }}"
+      availability_zones: "{{ item_c.zones }}"
+      vpc_zone_identifier: "{{ item_c.subnets }}"
+      tags: "{{ instance_tags }}"
 
-- name: Create Auto Scaling Group
```
Comment (mtulio) [inline tasks/cluster.yml]: @fabiorphp I did not find the retro compability here, we have another projects that uses this role with default behavior (using `cluster=ec2` - but it is not defined, beacuse is new option), if them update then run, I could see an broken execution, right?

What do you think? I believe that we can leva the default behavior in `ec2` and declare `fargate`.

File: README.md [hunk 4]
```diff
@@ -28,7 +28,23 @@ ecs_clusters:                                     # the list of ECS cluster conf
       - us-east-1a
     subnets:                                      # List of subnets
       - subnet-abcd1234
+    cluster_type: ec2                             # Cluster type: ec2 or fargate
+
```
Comment (mtulio) [inline README.md]: It could be clear if some comments here, like: "To create an fargate cluster you just run...."

**Review by mtulio — CHANGES_REQUESTED** [review]

File: tasks/cluster.yml (hunk 3 above)
Comment (fabiorphp) [reply tasks/cluster.yml]: Makes sense

Comment (fabiorphp) [reply tasks/cluster.yml]: done!

File: README.md (hunk 4 above)
Comment (fabiorphp) [reply README.md]: done!

**Comment by fabiorphp** [comment]
Any news?

**Comment by fabiorphp** [comment]
Guys! I don't whats happen but I don't have permissions to push.

**Review by mtulio — APPROVED** [review]
LGTM. Please, rebase.

We should test the both cases:
- launch fargate cluster
- launch ec2 cluster

Unfortunately we dont have CI tests to simulate it. :(

I believe that now we have retrocompability.

Very thanks for the contribution.

**Comment by fabiorphp** [comment]
Rebased! Can I merge?

**Pull request merged**